### PR TITLE
Remove telemetry

### PR DESCRIPTION
**Pull request by kevinguo455**
### Linked issue(s)
https://linear.app/kolena/issue/KOL-6307/remove-telemetry-from-python-sdk

### What change does this PR introduce and why?
Removes client telemetry. We do not get value out of this and it adds complexity to Python client.

### Please check if the PR fulfills these requirements

- [x] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [ ] Relevant docs have been added / updated

### CODE REVIEW

File: tests/unit/test_krequests.py [hunk 1]
```diff
@@ -51,10 +50,10 @@ def clean_client_state() -> Iterator[None]:
         ({"timeout": (1, 60)}, None, None),  # attempt to override default args
         ({}, {"http": "dummy-proxy"}, {}),
         ({}, {}, {"some-header-key": "some-header-val"}),
-        ({}, {}, {"X-Kolena-Telemetry": "off"}),  # attempt to override default headers with client state
-        ({"headers": {"X-Kolena-Telemetry": "off"}}, {}, {}),  # attempt to override default headers with kwargs
+        ({}, {}, {}),  # attempt to override default headers with client state
```
Comment (ly-kolena) [inline tests/unit/test_krequests.py]: Let's update these cases with new values to preserve the original purposes.

File: kolena/_utils/state.py [hunk 2]
```diff
@@ -24,9 +24,8 @@
 
 import requests
 
-import kolena
```
Comment (ly-kolena) [inline kolena/_utils/state.py:27]: What's the reason for this change? There are a couple usages in this file with `kolena.xxx`. Let's revert the change if it is not needed.

Comment (kevinguo455) [reply kolena/_utils/state.py:27]: The pre-commit hooks removed this line. Now that I look at it, I'm not entirely sure why the `kolena.xxx` usages still work with `import kolena` removed.

**Review by ly-kolena — APPROVED** [review]

File: kolena/_utils/state.py [hunk 3]
```diff
@@ -153,7 +146,7 @@ def get_token(
         proxies=proxies,
     )
     try:
-        krequests.raise_for_status(r)
+        kolena._utils.krequests.raise_for_status(r)
```
Comment (yifanwu-kolena) [inline kolena/_utils/state.py:149]: this [commit](https://github.com/kolenaIO/kolena/pull/594/commits/79a5d59adc77b1b2c8a53e4fba7dd77f4e22abf3) againsts with here, might need to revert this diff and add back `from kolena._utils import krequests` as before to keep the explicit importing aligning with the usage in this file

Comment (yifanwu-kolena) [reply kolena/_utils/state.py:149]: nvm, my comment is invalid
